### PR TITLE
80 notification backend add count notification computation logic

### LINKED ISSUE #80: Notification backend: add count notification computation logic
Read the dashboards_config.json. It contains the information which LAPIS instance to call for which organism.

Implement the trigger logic for the "count" trigger:
The trigger contains a window to look at (e.g. last two months)
Compute the dateTo and dateFrom depending on today's date
Call the respective LAPIS instance with the trigger filter and the computed dateTo and dateFrom.

The subscription should contain the information which date column to look at.

### PR DESCRIPTION
**Pull request by fengelniederhammer**
resolves #80

### CODE REVIEW

File: backend/src/main/kotlin/org/genspectrum/dashboardsbackend/api/TriggerEvaluation.kt [hunk 1]
```diff
@@ -16,15 +16,23 @@ sealed interface TriggerEvaluationResult {
         }
     }
 
-    data object ConditionMet : TriggerEvaluationResult {
+    data class ConditionMet(
+        val actualResult: Int,
```
Comment (JonasKellerer) [inline backend/src/main/kotlin/org/genspectrum/dashboardsbackend/api/TriggerEvaluation.kt:20]: I'm not happy with the naming. How about `evaluatedValue`?

File: backend/src/test/kotlin/org/genspectrum/dashboardsbackend/controller/SubscriptionsControllerTriggerEvaluationTest.kt [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    @Autowired private val subscriptionsClient: SubscriptionsClient,
+) {
+    @MockkBean
+    private lateinit var lapisClientProviderMock: LapisClientProvider
+
+    @Test
+    fun `GIVEN lapis returns count greater than threshold WHEN evaluating trigger THEN returns condition met`() {
+        val userId = getNewUserId()
+
+        val createdSubscription = subscriptionsClient.postSubscription(dummySubscriptionRequest, userId)
```
Comment (JonasKellerer) [inline backend/src/test/kotlin/org/genspectrum/dashboardsbackend/controller/SubscriptionsControllerTriggerEvaluationTest.kt:39]: No serious complaint, but can we somehow overwrite the value of the trigger thershold here from the dummySubscriptionRequest through a copy, or something similar? Only from reading the test it is not clear, why this test should pass and the dummyRequest lies in another file.

File: backend/src/main/resources/db/migration/V1__init.sql [hunk 3]
```diff
@@ -3,10 +3,8 @@ create table subscriptions_table (
     name text not null,
     interval VARCHAR(255) not null,
     active boolean not null,
-    conditions_met boolean not null,
```
Comment (JonasKellerer) [inline backend/src/main/resources/db/migration/V1__init.sql:6]: Just a comment: When we do it this, we need to call in the frontend the `/evaluateSubscription` endpoint for each subscription the user has on the `/subscriptions` frontend page. Is this what we want to do? Should the user be shown the current values, or the stored values of the trigger with an additional button to evaluate the triggers?

**Review by JonasKellerer — APPROVED** [review]
Nice work.
I think we need to discuss how the subscriptions should be evaluated in the frontend. This way one needs to call for each subscription on the subscription page the endpoint and get the result.

File: backend/src/main/resources/db/migration/V1__init.sql (hunk 3 above)
Comment (fengelniederhammer) [reply backend/src/main/resources/db/migration/V1__init.sql:6]: As discussed: We can still add it again later. Latest when we implement notifications via some channel, we probably need to store the last result anyway.

**Pull request merged**